sdk/python/src/freewayai/system_logger.py

```diff
@@ -0,0 +1,64 @@
+import json
+from .llm_system import LLMSystem
+from .prompt_template import PromptTemplate
+
+class SystemLogger:
+    def __init__(self, system : LLMSystem, log_location=None) -> None:
+        self.system = system
+        if log_location is None or isinstance(log_location, str) or hasattr(log_location, 'write'):
+            self.log_location = log_location
+        else:
+            raise ValueError("log_location must be a string or a file pointer or None")
+
+    def log_llm_interaction(self, prompt=None, response=None, prompt_id:str=None, prompt_template:PromptTemplate=None, model:str=None, variables:dict=None):
+        # Format a json body containing each added variable plus the system.system_id
+        log_body = dict()
```

Comment:
This is all fine but as I'm reading it, it seems like the more common thing to do here would be to have an instance of a logger, namespaced properly, called here. You would pass all of the values to [extra](https://docs.python.org/3/library/logging.html#loggeradapter-objects) in the logger. We could talk about it and update that in a future version though.

Reply:
Yup. We can tackle later, agreed